src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -161,11 +163,11 @@ public void initialize() {
      */
     public boolean shutdown() throws InterruptedException {
         logger.info("JobService received shutdown request");
-        if (jobShutdownState.isShutdown()) {
+        if (isShutdown.get()) {
             logger.warn("Ignoring duplicate shutdown request");
             return true; // allow this to be success
         }
-        jobShutdownState.setShutdown();
+        isShutdown.set(true);
```

Comment:
I'm not very familiar with AtomicBoolean, but "isShutdown" seems like a good use case for it. From a quick google to learn about it, I'm wondering if we could use getAndSet(true) here? It's not really any cleaner, but I think it would actually take advantage of the "atomic" characteristic of the AtomicBoolean? Thoughts?

So,  I think it would look something like the following. 
```
boolean currentlyShutdown = isShutdown.getAndSet(true);
if(currentlyShutdown){
            logger.warn("Ignoring duplicate shutdown request");
            return true; // allow this to be success
}
```

Reply:
Yes, that is an improvement.